src/ir/item.rs

```diff
@@ -1444,6 +1445,15 @@ impl ItemCanonicalPath for Item {
                                       ctx: &BindgenContext)
                                       -> Vec<String> {
         let path = self.canonical_path(ctx);
+        if let super::item_kind::ItemKind::Type(ref type_) = self.kind {
+            if let &TypeKind::Enum(ref enum_) = type_.kind() {
```

Comment:
nit: `if let TypeKind::Enum(ref enum_) = *type.kind()` looks more consistent with the rest of the code.